### PR TITLE
Adding custom environment variables

### PR DESCRIPTION
**Pull request by royalrick**
### Background

1. For areas where Google is blocked, a proxy is needed to access Google.

2. Third-party vendors similar to [API2D](https://api2d.com) have emerged for Openai's interface, and their interfaces may be more suitable for certain regions. If you want to use their interface, you need to replace Openai's restricted address.

### Changes

1. add environment variables 'HTTP_PROXY_HOST' and 'HTTP_PROXY_PORT'  for http proxy.
2. add environment variable 'OPENAI_API_BASE' for custom Openai api's base address.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by richbeales** [comment]
please re-title

File: .env.template [hunk 1]
```diff
@@ -141,3 +144,13 @@ TW_CONSUMER_KEY=
...
+
+################################################################################
+### PROXY
+################################################################################
+
+### HTTP PROXY
+# HTTP_PROXY_HOST - HTTP proxy host (Example:127.0.0.1)
+# HTTP_PROXY_PORT - HTTP proxy port (Example: 7890)
+HTTP_PROXY_HOST=127.0.0.1
+HTTP_PROXY_PORT=7890
```
Comment (nponeccop) [inline .env.template]: The final newline is missing

**Review by nponeccop — DISMISSED** [review]

File: .env.template (hunk 1 above)
Comment (royalrick) [reply .env.template]: already resolved.

**Review by nponeccop — CHANGES_REQUESTED** [review]
Change the PR title to something meaningful as requested by @richbeales

**Comment by royalrick** [comment]
> Change the PR title to something meaningful as requested by @richbeales

OK

**Review by nponeccop — DISMISSED** [review]

**Comment by nponeccop** [comment]
@richbeales He fixed

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Pwuts** [comment]
Thanks for submitting! This PR may be old, but it's not unmergeable.

A few remarks though:
* Implementing measures to circumvent internet blockages is not within Auto-GPT's scope.
* The possibility to set custom API base URLs was added with #2594 

That doesn't really leave anything still to be merged in this PR. Still, feel free to join our discord and weigh in, discuss or join in on the action there! Useful contributions are always welcome, even if we sometimes take a while to review & merge. :)

**Pull request closed without merging**